### ISSUE TITLE
[BUG] Cloud Updated Pilots losing assigned group in Pilot Roster

### ISSUE THREAD

**Issue by compcon** [massif-press/compcon#988]
<strong>Submitted At: </strong>7/19/2020 12:44:54
<strong>Platform: </strong>
Browser - Chrome
<strong>Describe the bug: </strong>
When a pilot is updated from the cloud, they lose their assigned group in the Pilot Roster, becoming ungrouped
<strong>To Reproduce: </strong>
Assign a Pilot made by another user to a Custom Group. Then update the Pilot using either Quick Sync or clicking Update From Cloud by the Omninet Uplink Id in  the Dossier section. The pilot will return to the Ungrouped group.
<strong>Expected behavior: </strong>
A Pilot updated from the cloud retaines the group I have assigned them locally for my campaign
<strong>Additional context: </strong>
Before: https://cdn.discordapp.com/attachments/619659613322608650/734464390690111558/unknown.png 
Update from Cloud: https://cdn.discordapp.com/attachments/619659613322608650/734464659670695957/unknown.png
After: https://cdn.discordapp.com/attachments/619659613322608650/734464720773316688/unknown.png

This is so minor it isn't even an inconveniance -- just thought I should make you aware. I couldn't find it in the open bug list already. (I may be blind). If it's an intended feature, just nuke this report.

Thank you Comp/Con team for all the hard work ( ' u ' )

<strong>Contact Info: </strong>
Fibby#6376
<strong>Stack Trace: </strong>

**Comment by RobotKernel** [comment]
Quickly looked at this. This "bug" appears because the pilot's group is saved to gist. If you load another user's pilot and they have assigned it to one of their own groups then that group will also appear on your pilot roster. If the original user didn't set a group it resets it to ungrouped on sync. Question is if it's functioning as intended. Users might want to send their group along when sharing their pilot. Especially if the user is sharing across their own devices. On the other hand GM's probably want to use their own groups instead of their players' groups.

My suggestion would be to only sync groups if the Pilot is user owned. 
Seeing as this is a cloud load only issue I would suggest a solution along the lines of:
```es6
  public async CloudLoad(): Promise<any> {
    if (!this.CloudID) return Promise.reject('No Cloud ID')
    return gistApi.loadPilot(this.CloudID).then((gist: IPilotData) => {
      if(!this.IsUserOwned) {
        gist.group = this._group
      }
      this.setPilotData(gist)
      this.LastCloudUpdate = new Date().toString()
    })
  }
```

Will post a pull request with this fix.

**Comment by jarena3** [comment]
Corrected in 2.2.6
